doc/HACKING.md

```diff
@@ -0,0 +1,27 @@
+# Hacking on the kata-operator
+
+## Using a custom kata-operator-payload image
+
```

Comment:
Do you think it makes sense to add the "why" (probably copied from the issue description) we would want to set a custom image?

Reply:
It sure doesn't hurt :-) :+1: